api/api.py

```diff
@@ -361,36 +394,55 @@ class CI_Measurement(BaseModel):
 
 @app.post('/v1/ci/measurement/add')
 async def post_ci_measurement_add(measurement: CI_Measurement):
-    # error_helpers.log_error(error_message)
-    for i in CI_Measurement.schema()['properties'].keys():
-        key_value = getattr(measurement, i)
-        if i == 'project_id':
-            if key_value is None or key_value.strip() == '':
+    for key, value in measurement.dict():
+        if key == 'project_id':
+            if value is None or value.strip() == '':
                 measurement.project_id = None
-            elif not is_valid_uuid(key_value.strip()):
-                return {'success': False, 'err': "project_id is not a valid uuid"}
-        elif i == 'label':
-            if key_value is None or key_value.strip() == '':
+                continue
+            elif not is_valid_uuid(value.strip()):
+                return {'success': False, 'err': f"project_id '{value}' is not a valid uuid"}
+            else:
+                measurement.project_id = escape(value)
+                continue
+        if key == 'label':
+            if value is None or value.strip() == '':
                 measurement.label = None
-        elif i == 'value':
-            if key_value is None:
-                return {'success': False, 'err': f"{i} is empty"}
-        elif i == 'unit':
-            if key_value is None or key_value.strip() == '':
-                return {'success': False, 'err': f"{i} is empty"}
-            if key_value != 'mJ':
+            else:
+                measurement.label = escape(value)
+            continue
+        if key == 'unit':
+            if value is None or value.strip() == '':
+                return {'success': False, 'err': f"{key} is empty"}
+            if value != 'mJ':
                 return {'success': False, 'err': "Unit is unsupported - only mJ currently accepted"}
-        elif key_value is None or key_value.strip() == '':
-            return {'success': False, 'err': f"{i} is empty"}
+
+        if value is None or value.strip() == '':
+            return {'success': False, 'err': f"{key} is empty"}
+
+        if key == 'repo':
+            measurement.repo = escape(value)
+            continue
+        if key == 'branch':
+            measurement.branch = escape(value)
+            continue
+        if key == 'workflow':
+            measurement.workflow = escape(value)
+            continue
+        if key == 'run_id':
+            measurement.run_id = escape(value)
+            continue
+        if key == 'source':
+            measurement.source = escape(value)
+            continue
```

Comment:
This could be rewritten to be
```
else_keys = ['repo', 'branch', 'workflow', 'run_id', 'source']
if key in else_keys:
        setattr(measurement, key, escape(value))
```

or you could use:
```
    match value:
        case 'case1' | 'case2':
            return 'This is case 1 or 2'
        case 'case3':
            return 'This is case 3'
        case _:
            return 'This is default case'

```
for the whole block